harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -44,22 +44,20 @@ def __init__(self, trial_inst: det.Trial, *args: Any, **kwargs: Any) -> None:
             if util.is_overridden(callback.on_checkpoint_end, pytorch.PyTorchCallback):
                 warnings.warn(
                     "The on_checkpoint_end callback is deprecated, please use "
-                    "on_checkpoint_write_end instead",
+                    "on_checkpoint_write_end instead.",
                     FutureWarning,
                 )
 
-        check.gt_eq(
-            len(self.context.models),
-            1,
-            "Must have at least one model. "
-            "This might be caused by not wrapping your model with wrap_model()",
-        )
-        check.gt_eq(
-            len(self.context.optimizers),
-            1,
-            "Must have at least one optimizer. "
-            "This might be caused by not wrapping your optimizer with wrap_optimizer()",
-        )
+        if not self.context.models:
+            raise det.errors.InvalidExperimentException(
+                "Must have at least one model. "
+                "This might be caused by not wrapping your model with wrap_model().",
+            )
+        if not self.context.optimizers:
```

Comment:
another case of truthy check instead of length check.

Reply:
Fixed.